ingestor-default/src/test/scala/za/co/absa/hyperdrive/ingestor/implementation/utils/TestKafkaUtilDockerTest.scala

```diff
@@ -185,20 +165,22 @@ class TestKafkaUtilDockerTest extends FlatSpec with Matchers with BeforeAndAfter
     exception.getMessage should include ("Requested consumption")
   }
 
-  "getAtLeastNLatestRecords" should "get at least the n latest records" in {
```

Comment:
I am a bit confused here. 
Why "segment.ms" -> s"$messageCreationTimeout", retention will kick off every 100 ms 
Why you have "timeout.foreach(Thread.sleep)"  in "produceData" method when you wait in test "    Thread.sleep(waitForCompactionMillis)"

Reply:
Without the sleep in `produceData`, all the messages would end up in the active segment, and the active segment doesn't get compacted.